doc/whats_new/v1.3.rst

```diff
@@ -41,6 +41,10 @@ Changelog
 - |Feature| :class:`pipeline.FeatureUnion` can now use indexing notation (e.g.
   `feature_union["scalar"]`) to access transformers by name. :pr:`25093` by
   `Thomas Fan`_.
+- |Feature| :class:`pipeline.FeatureUnion` can now access the
+  `feature_names_in_` attribute if the value seen during `.fit` was a `pandas
+  .DataFrame`.
```

Comment:
looking [here](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/validation.py#L1893), the check is not whether it is pandas dataframe, but whether an attribute `columns` is present. 

Maybe being more precise would help. (although most users tend to only use pandas/numpy with scikit-learn)

Reply:
or we could say what was written when this feature was introduced 
https://github.com/scikit-learn/scikit-learn/blame/main/doc/whats_new/v1.1.rst#L261

Reply:
@Kshitij68 I've updated the changelog. What do you think?

Reply:
Since the change log is user facing and most people will be using a DataFrame, I prefer saying a pandas DataFrame. 

Implementation-wise we check for `columns`, because scikit-learn does not want a hard dependency on pandas. Although, technically we support "array-likes" with a `columns` attribute with all strings, the primary focus of the feature is for DataFrames.

Reply:
> Since the change log is user facing and most people will be using a DataFrame, I prefer saying a pandas DataFrame.
> 
> Implementation-wise we check for `columns`, because scikit-learn does not want a hard dependency on pandas. Although, technically we support "array-likes" with a `columns` attribute with all strings, the primary focus of the feature is for DataFrames.

Should I revert back to the previous? Or add `pandas.DataFrame` to the description? Something like

```rst
- |Feature| :class:`pipeline.FeatureUnion` can now access the
  `feature_names_in_` attribute if the `X` value seen during `.fit` has a
  `columns` attribute and all columns are strings. e.g. when `X` is a 
  `pandas.DataFrame`.
  :pr:`25220` by :user:`Ian Thompson <ianiat11>`.
```